src/system_reader.rs

```diff
@@ -336,7 +415,16 @@ impl<R: RawReader> SystemReader<R> {
             // This is not an append. Clear the current symbol table.
             self.symbol_table.reset();
         }
-        // TODO: support importing shared symbol tables
+        // This for loop consumes the `String` values, clearing `self.lst.imported_symbols`
+        for value in self.lst.imported_symbols.drain(..) {
+            if let Some(text) = value {
+                // This symbol has defined text. Add it to the symbol table.
+                self.symbol_table.intern(text);
+            } else {
+                // This symbol was a null or non-string value. Add a placeholder.
+                self.symbol_table.add_placeholder();
```

Comment:
`SymbolTable` has a method called `intern_or_add_placeholder` that you can use for this:

https://github.com/amazon-ion/ion-rust/blob/fd09e870e5206be16a1c535da0abdd450c6ea6b5/src/symbol_table.rs#L71-L78